app/experimenter/docs/adrs/0003-analysis-visualization-architecture-approach.md

```diff
@@ -0,0 +1,49 @@
+# Analysis Visualization: Architecture Approach
+
+* Status: accepted
+* Author: Marina Samuel
+* Deciders: Nimbus Team
+* Date: 2020-09-16
+
+## Context and Problem Statement
+
+Analysis and visualization of experiment result data needs to be [part of the experiment console](https://github.com/mozilla/nimbus-shared/blob/main/docs/adr/0001-one-console-app.md). The mockups based on user interviews are [here](https://www.figma.com/file/q6Xh9g6LrOj2OQ59LbAFXm/Experiment-Dashboard-Mockup).
+
+Some decisions need to be made about how the data will be pulled out of [Jetstream](https://github.com/mozilla/jetstream) and visualized within the console. This ADR will highlight how this will be done.
+
+
+## Decision Drivers
+
+Drivers for fetching data:
+* Data is easy to fetch and minimizes the amount of new state added
+
+Drivers for visualizing data:
+* Reuse existing [Graph Paper](https://github.com/graph-paper-org/graph-paper) UI components if possible
+* How easy or possible is it to integrate Svelte components into React (since Graph Paper is in Svelte)
+
+
+## Considered Options
+
+**Fetching Data**
+1. Add a POST endpoint to Jetstream to send data over to Experiment Console. This data would be saved in the Django database and fetched from the front-end when needed using GraphQL.
+2. Add a REST API endpoint to Experiment Console which fetches the Jetstream data using GCS dependencies. These are fetched on-demand and not stored in the Django database.
+
+**Frontend**
+1. Use nvd3 or other graphing library as an alternative to Graph Paper
+2. Use Graph Paper/Svelte only for graphs, otherwise, all visualizations from the mockup will be in React
+3. Use Svelte for the entire visualization portion of the console
+
+## Decision Outcome
+
+**Fetching Data**
+Option (2) was chosen.
+
+While option (2) requires adding more state such as bucket credentials, it is more straight-forward than option (1). It does not require any changes on the Jetstream side to make POST calls. It also does not require storing the data to the Django database. This is ok since the aggregated data is very small, in the range of around 100kb per fetch.
+
```

Comment:
Giving write access to Jetstream would also change the authentication requirements and require a security review etc.